.github/workflows/update-schema.yml

```
@@ -81,15 +81,106 @@ jobs:
         run: |
           mage -v schema:readme ${{ github.event.inputs.ref }}
 
-      - name: Setup node environment
+      - name: Upload artefact
+        uses: actions/upload-artifact@v4
+        with:
+          name: schema
+          path: ./
+          retention-days: 1
+          overwrite: true
+
+  schema-matrix:
+    runs-on: ubuntu-22.04
+    needs: update-schema
+    outputs:
+      schema: ${{ steps.set-matrix.outputs.schema }}
+    steps:
+      - uses: actions/checkout@v4
+      - id: set-matrix
+        run: echo "::set-output name=schema::$(ls schema/ | jq -R -s -c 'split("\n")[:-1]' )"
+
+  build-target:
+    runs-on: ubuntu-22.04
+    needs: schema-matrix
+    strategy:
+      matrix:
+        target: [ts, go]
+        schema: ${{ fromJson(needs.schema-matrix.outputs.schema) }}
+      max-parallel: 1
+    steps:
+      - name: Download artefact
+        uses: actions/download-artifact@v4
+        with:
+          name: schema
+          path: ./
+
+      - name: Setup node environment (for building)
         uses: actions/setup-node@v4
         with:
-          node-version: 18.12
+          node-version: 18.12.0
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22"
+          cache: true
+          cache-dependency-path: |
+            **/go.sum
+            **/go.mod
+
+      - name: Install Mage
+        run: go install github.com/magefile/mage@v1.15
+
+      - name: Installing quicktype
+        run: |
+          yarn global add quicktype
```

Please specify version for `quicktype`.

```suggestion
          yarn global add quicktype@23
```